demo

```diff
@@ -62,16 +80,138 @@ if [[ "$?" -eq 1 ]]; then
+
+docker-compose \
+    -f docker-compose.yml \
+    -f docker-compose.storage.minio.yml \
+    -f docker-compose.demo.yml \
+    $CLIENT \
+    "$@" -d
+
+USER='mender-demo@example.com'
+PASSWORD=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c12)
+
+MENDER_SERVER_USER="${USER}:${PASSWORD}"
+MENDER_SERVER_URI='https://localhost'
+
+# Block until the useradm service returns an HTTP response.
+# Which means that it is up.
+echo "Waiting for the useradm service to start up"
+curl --silent -k -X POST -u $MENDER_SERVER_USER\
+     --fail\
+     --retry-connrefused\
+     --retry 60\
+     --retry-delay 2\
+     --retry-max-time 120\
+     $MENDER_SERVER_URI/api/management/v1/useradm/auth/login > /dev/null
+
+retval=$?
+case $retval in
+    0)  ;; # User exists - continue.
+    22) ;; # Server 400 error, ie, server is up - continue.
+    *) echo "Failed to access the mender-useradm service"
+       echo "Please check the docker-compose setup."
+       echo "curl exit code" $retval
+       exit $retval
+       ;;
+esac
+
+echo "Creating a new user..."
+
+# Create a new user.
+docker exec ${DOCKER_COMPOSE_PROJECT_NAME}_mender-useradm_1 /usr/bin/useradm create-user --username=${USER} --password=${PASSWORD}
+
+retval=$?
+case $retval in
+     0) echo "New user created." ;;
+     5) echo "The user already exists"; exit 1;; # No need to continue if a demo user exists
+     *) exit $retval ;;
+esac
+
+echo "Getting JWT Token..."
+
+JWT=$(curl --silent -k -X POST -u $MENDER_SERVER_USER\
+           --fail\
+           --retry-connrefused\
+           --retry 60\
+           --retry-delay 2\
+           --retry-max-time 120\
+           $MENDER_SERVER_URI/api/management/v1/useradm/auth/login)
+
+retval=$?
+if [ $retval -ne 0 ]; then
+    echo "Failed to get the 'JWT' token from the useradm service."
+    echo "This is needed in order to use the demo Artifact."
+    exit $retval
+fi
+
+echo "Uploading the artifact to the server..."
+errout=$(curl -k --request POST\
+              --silent\
+              --show-error\
+              --retry-connrefused\
+              --retry-delay 2\
+              --retry 60 \
+              --retry-max-time 120\
+              --header "Authorization: Bearer ${JWT}"\
+              --form "size=${ARTIFACT_SIZE_BYTES}"\
+              --form "artifact=@${DEMO_ARTIFACT_NAME}"\
+              "${MENDER_SERVER_URI}/api/management/v1/deployments/artifacts" | jq .error)
+
+retval=$?
+
+case "$errout" in
+   " Artifact not unique"*) echo "Artifact already exists on the server" ;;
+    "") echo "Artifact uploaded to the demo server" ;;
+    *) echo "Uploading the demo artifact failed with error: " $errout
+       exit $retval ;;
+esac
+
+echo "****************************************"
+echo
+echo  "Login password: ${PASSWORD}"
+echo
+echo "****************************************"
+echo "Please keep the password available, it will not be cached by the login script."
+echo "Backend ready. Press Enter to show logs or Ctrl-C to quit."
```

Comment:
So my point from yesterday demo.

The verb "quit" here IMO is not clear about what it will do. My first read is like now is ready, running in the background (because I don't see anything in the console), so I can "quit" the script and go to the website in my Firefox.

For the text to be a bit clearer, I suggest something like: "Backend ready and running in the background. Pres Enter to show logs. Pres Ctrl-C to stop the backend and quit."

I leave it up to you, but one important thing to me is to separate the actions with something more than "or".

Reply:
+1 Incorporated